File: ios/Sources/Detonator/Request/VideoSeekRequest.swift
import AVFoundation

class VideoSeekRequest: Request {
    override public func run() {
        let position: Int! = decodeData()
        
        let edge = getComponentEdge()!
        
        let videoEdge = edge.children[0]
        
        let element = videoEdge.element as! VideoElement
        
        let seconds = Double(position) / 1000
    
        let time = CMTime(seconds: seconds, preferredTimescale: 600)
        
        element.player?.seek(to: time)
        
        end()
    }
}

File: android/src/main/java/com/iconshot/detonator/request/VideoSeekRequest.java
package com.iconshot.detonator.request;

import androidx.media3.common.Player;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.element.videoelement.VideoElement;
import com.iconshot.detonator.renderer.Edge;

public class VideoSeekRequest extends Request<Integer> {
    public VideoSeekRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    public void run() {
        Integer position = decodeData(Integer.class);

        Edge edge = getComponentEdge();

        Edge videoEdge = edge.children.get(0);

        VideoElement element = (VideoElement) videoEdge.element;

        Player player = (Player) element.player;

        if (player != null) {
            player.seekTo(position);
        }

        end();
    }
}
